frontend/lib/src/util/UriUtil.ts

```diff
@@ -69,29 +71,52 @@ export function getWindowBaseUriParts(): BaseUriParts {
 // easy solution covering every deployment scenario.
 export function getPossibleBaseUris(): Array<BaseUriParts> {
   const baseUriParts = getWindowBaseUriParts()
+  const configBasePath: string | undefined = getBaseUrlPath()
+  const possibleBaseUris: Array<BaseUriParts> = []
+  const possibleBasePaths: Set<string> = new Set()
+
+  if (configBasePath !== undefined) {
+    const cleanedConfigBasePath = configBasePath
+      .replace(FINAL_SLASH_RE, "")
+      .replace(INITIAL_SLASH_RE, "")
+    possibleBaseUris.push({
+      ...baseUriParts,
+      basePath: cleanedConfigBasePath,
+    })
+
+    possibleBasePaths.add(cleanedConfigBasePath)
+  }
+
   const { basePath } = baseUriParts
 
-  if (!basePath) {
-    return [baseUriParts]
+  if (!basePath && !possibleBasePaths.has(basePath)) {
+    return possibleBaseUris.concat([baseUriParts])
   }
 
   const parts = basePath.split("/")
-  const possibleBaseUris: Array<BaseUriParts> = []
 
   while (parts.length > 0) {
+    const partialBasePath = parts.join("/")
+    parts.pop()
+
+    if (possibleBasePaths.has(partialBasePath)) {
+      continue
+    }
+
     possibleBaseUris.push({
       ...baseUriParts,
-      basePath: parts.join("/"),
+      basePath: partialBasePath,
     })
-    parts.pop()
   }
 
-  possibleBaseUris.push({
-    ...baseUriParts,
-    basePath: "",
-  })
+  if (!possibleBasePaths.has("")) {
```

Comment:
It might make sense to add small comments to each of these steps to explain the check and the action (aka why do we check for `""` here / what does it mean?)